.github/actions/first-time-contributor/entrypoint.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -e
+
+# 1. Get the author and pr number for the pull request.
+author=$(jq -r '.pull_request.user.login' $GITHUB_EVENT_PATH)
+pr_number=$(jq -r '.number' $GITHUB_EVENT_PATH)
+
+if [ $pr_number = "null" ] || [ $author = "null" ]; then
+    echo "Could not find PR number or author. $pr_number / $author"
+	exit 78
+fi
+
+# 2. Fetch the author's commits list for the repo to determine if they're a first-time contributor.
+echo "Fetching commits for using the url https://github.com/$GITHUB_REPOSITORY/commits?author=$author"
+html=$(
+	curl \
+		--silent \
+		"https://github.com/$GITHUB_REPOSITORY/commits?author=$author"
+)
+
+# 3. Run a regular expression against the html to check for the 'No commits found' message.
+regex='\<p\>[[:space:]]*No[[:space:]]+commits[[:space:]]+found[[:space:]]+for[[:space:]]+"'$author'"[[:space:]]*\<\/p\>'
+if ! [[ "$html" =~ $regex ]]; then
```

Comment:
This regular expression works on my local command line, but not when run in docker. Not sure the best way to fix it.

Reply:
Have resolved this now. Instead of scraping from an HTML page I'm using the github search api to fetch contributions. Props to @pento for the suggestion!